nixos/modules/system/boot/networkd.nix

```diff
@@ -641,13 +640,13 @@ let
       '';
     };
 
-    dhcpConfig = mkOption {
+    dhcpV4Config = mkOption {
```

Comment:
We probably still want to add a `mkRemovedOptionModule`, and print a short summary of what's in the release notes instead of just dropping the configuration options here.

Reply:
I tried adding that but due to the nature of how those work that isn't really feasible here. We have a "wildcard" network name in between and can only really assert if someone tries to access that value.

I tried a few things:

- traversing the `config` tree to see if the option is set, that does then trigger the "on access" assertion
- traversing the `option` tree (as it is done in the original helper) but that doesn't seem possible for these kinds of situations..

Here is the code I ended up backing out of this changest in case anyone wants to give it a shot:

```nix
# systemd.network.networks.*.dhcpConfig has been deprecated in favor of ….dhcpV4Config
# Produce a nice warning message so users know it is gone.
{
  assertions = let
    #affectedNetworks = attrNames (filterAttrs (network: conf: conf.dhcpConfig.isDefined) options.systemd.network.networks.values);
    affectedNetworks = attrNames (filterAttrs (network: conf: conf ? dhcpConfig) config.systemd.network.networks);
  in
  [
    {
      assertion = length affectedNetworks == 0;
      message = ''
        The option definition `systemd.network.networks.*.dhcpConfig` has been deprecated in favor of `systemd.network.networks.*.dhcpV4Config`. Please update your configuration.
        The following networks are still using it:
        ${concatMapStringsSep "\n" (network: "  - `systemd.network.networks.\"${network}\"`") affectedNetworks}
      '';
    }
  ];
}

```

Reply:
For now I just left the "on access" throw in the code as that is the least we could do. I do not really want to make it an alias.